src/sinks/aws_kinesis_firehose.rs

```diff
@@ -461,15 +461,14 @@ mod integration_tests {
         let hits = response["hits"]["hits"]
             .as_array()
             .expect("Elasticsearch response does not include hits->hits");
-        let input = input
+        let mut input = input
             .into_iter()
-            .map(|rec| serde_json::to_value(&rec.into_log()).unwrap())
-            .collect::<Vec<_>>();
+            .map(|rec| serde_json::to_value(&rec.into_log()).unwrap());
         for hit in hits {
             let hit = hit
                 .get("_source")
                 .expect("Elasticsearch hit missing _source");
-            assert!(input.contains(hit));
+            assert!(input.any(|e| &e == hit));
```

Comment:
What is the change to `contains` that caused this? I didn't see it listed in the release notes.

Reply:
Ah, this is actually clippy complaining that we can just use the original iterator rather than collecting it first.

https://rust-lang.github.io/rust-clippy/master/#needless_collect

Reply:
Ok, right, I saw that, which is a great lint to have around, but forgot that `trait Iterator` doesn't have a `fn contains`.

Reply:
Agreed. There is actually one in `itertools` which could be used too.